flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/runtime/stream/sql/ProcedureITCase.java

```diff
@@ -210,6 +214,31 @@ void testNamedArgumentsWithOptionalArguments() {
                 ResolvedSchema.of(Column.physical("result", DataTypes.STRING())));
     }
 
+    @Test
+    void testEnvironmentConf() throws DatabaseAlreadyExistException {
+        // root conf should work
+        Configuration configuration = new Configuration();
+        configuration.setString("key1", "value1");
+        StreamExecutionEnvironment env =
+                StreamExecutionEnvironment.getExecutionEnvironment(configuration);
+        StreamTableEnvironment tableEnv = StreamTableEnvironment.create(env);
+        TestProcedureCatalogFactory.CatalogWithBuiltInProcedure procedureCatalog =
+                new TestProcedureCatalogFactory.CatalogWithBuiltInProcedure("procedure_catalog");
+        procedureCatalog.createDatabase(
+                "system", new CatalogDatabaseImpl(Collections.emptyMap(), null), true);
+        tableEnv.registerCatalog("test_p", procedureCatalog);
+        tableEnv.useCatalog("test_p");
```

Comment:
Can you please also add a property ("key1, "value22") to table config , and then call the get_env_conf to make sure we can get both table configuration and root configuration

Reply:
if also add a property with key1, it will also overwrite the same key again. Do you mean to add a property ("key2", "value22") to table config and then check?

Reply:
That's right. Sorry for mistake.  I mean:
you use  "key1" = "value1" to create root configuration.
Then, set "key2" = "value2" to table configuration, then call the procedure.
to make sure it can get from both table configuration and root configuration

after that:
you set "key1" = "value11" to table configuration, 
then call the procedure.to make sure the table configuration will overwrite root configuration

Reply:
OK, I have refine the test case.